pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -264,6 +319,44 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 	return nil
 }
 
+// getDefaultGateway returns the default gateway address with lower priority using route and device
+// statedb tables and the provided default gateway configuration.
+func (r *NeighborReconciler) getDefaultGateway(defaultGateway *v2.DefaultGateway) (string, error) {
+	var defaultRoute netip.Prefix
+	switch defaultGateway.AddressFamily {
+	case "ipv4":
+		defaultRoute = ipv4Default
+	case "ipv6":
+		defaultRoute = ipv6Default
+	default:
+		return "", fmt.Errorf("invalid address family %s", defaultGateway.AddressFamily)
+	}
+	txn := r.DB.ReadTxn()
+	// get routes from statedb route table
+	// TODO: add RoutePrefixIndex Query to lookup routes by prefix
+	routes := r.routeTable.All(txn)
+	activeDefaultRoutes := []*tables.Route{}
+	for route := range routes {
+		// ignore routes that are not default routes or do not have a valid gateway
+		if !route.Gw.IsValid() || route.Dst != defaultRoute {
```

Comment:
Does this work if the Gw is IPv6 link-local address? cc: @rastislavs

Reply:
I haven't tested it with link-local address as default gateway. i will test that.

Reply:
Good point, for IPv6 link-local default GW to work as a peer in GoBGP, you would need to return the peering address with zone here (e.g. interface name configured as zone using `netip.Addr.WithZone`)

As for testing in containerlab, you could try configuring [Router Advertisements](https://docs.frrouting.org/en/latest/ipv6.html#router-advertisement) on FRR and auto-configuration on cilium nodes, but I am not sure about the exact config (you may need sysctl configs like `accept_ra`, `accept_ra_defrtr`).

Reply:
I tried to make this work, but, looks like `bgp listen range` cannot be used with link local addresses because they expect interface as well. 
I think someone requested for per-interface bgp listen range, but, it wasn't implemented.
https://github.com/FRRouting/frr/issues/9465

```
bgp listen range fe80::/10 peer-group CILIUM 
% Malformed listen range (link-local address)
```

for link local addresses, i think we should probably stick to bgp unnumbered that we can probably add in the future?

Reply:
I see, I think we have 2 options here:
1) implement the link-local default gateway properly here and test it with FRR somehow manually,
2) skip link-local gateway addresses (maybe with a waring) and document that such GW addresses are not supported for this auto-discovery mechanism.

Reply:
i think it will be better to have a warning and document it. 
if bgp listen range does not support it, then we cannot do it anyway so hacking it in frr to make it would not be the right approach probably.

Reply:
i added a warning message